pilot/pkg/model/jwks_resolver.go

```diff
@@ -345,13 +345,23 @@ func (r *JwksResolver) getRemoteContentWithRetry(uri string, retry int) ([]byte,
 		}
 		defer resp.Body.Close()
 
-		body, err := io.ReadAll(resp.Body)
+		// RFC7517 does not specify a maximum length, but unbounded input may lead to resource exhaustion.
+		const maxBytesToRead int64 = 1048577 // 1MiB + 1Byte
```

Comment:
> Could we test the actual jwks response size that istiod is able to handle?

This of course depends on the RAM available both in the control plane and sidecar. I don't have a test setup handy.
I also don't like this 1MiB limit myself but don't have a better idea on how to prevent resource exhaustion.